src/App/Platforms/Android/Autofill/CredentialHelpers.cs

```diff
@@ -285,10 +285,29 @@ public static async Task<string> ValidateCallingAppInfoAndGetOriginAsync(Calling
             }
         }
 
-        private static async Task<string> ValidateNativeAppAndGetOriginAsync(CallingAppInfo callingAppInfo, string rpId)
+        public static async Task<string> ValidateNativeAppAndGetOriginAsync(CallingAppInfo callingAppInfo, string rpId)
```

Comment:
About `public`, were you going to add unit tests?

Reply:
Changed to `private` given that for the moment we're not going to add unit tests for this as this would need the UT project to run with `net8.0-android` target framework.

Reply:
Now moved the logic to a core service `AssetLinksService` so we can unit test it.